### PR TITLE
Limit retries to fetch the runtime artifacts. 

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose

Adapter will no longer retry the runtime-artifacts endpoint endlessly for responses received with status 500. Instead of that it should stop after 10 retries. And there is log statement to indicate that the retry is stopped for the particular API.

### Issues

Fixes #

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by VajiraPrabuddhaka — APPROVED** [review]

File: adapter/internal/synchronizer/apis_fetcher.go [hunk 1]
```diff
@@ -234,6 +239,13 @@ func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string, envToD
 			// Keep the iteration still until all the envrionment response properly.
 			logger.LoggerSync.Errorf("Error occurred while fetching data from control plane for the API %q: %v. Hence retrying..", updatedAPIID, data.Err)
 			sync.RetryFetchingAPIs(c, data, sync.RuntimeArtifactEndpoint, true, queryParamMap)
+			retryCounter++
```
Comment (slahirucd7) [inline adapter/internal/synchronizer/apis_fetcher.go:242]: Can't we return after satisfying above condition ?

File: adapter/internal/synchronizer/apis_fetcher.go [hunk 2]
```diff
@@ -234,6 +239,13 @@ func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string, envToD
...
 		}
 	}
+
+	if !receivedArtifact {
+		// This logs statement is used to
```
Comment (slahirucd7) [inline adapter/internal/synchronizer/apis_fetcher.go:247]: line 247 seems mistakenly deleted....

**Review by renuka-fernando — APPROVED** [review]

File: adapter/internal/synchronizer/apis_fetcher.go (hunk 2 above)
Comment (VirajSalaka) [reply adapter/internal/synchronizer/apis_fetcher.go]: fixed. thanks for pointing it out

File: adapter/internal/synchronizer/apis_fetcher.go (hunk 1 above)
Comment (renuka-fernando) [reply adapter/internal/synchronizer/apis_fetcher.go:242]: I think @slahirucd7  didn't see the break in line 243 (`if data.Resp != nil {` block)

**Review by VajiraPrabuddhaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230526.11

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230526.11

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230526.11

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230529.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230529.3

**Pull request merged**